cores/arduino/USBSerial.cpp

```diff
@@ -83,6 +83,8 @@ size_t USBSerial::write(const uint8_t *buffer, size_t size)
       buffer += portion;
       // After storing data, start transmitting process
       CDC_continue_transmit();
+    } else if (!CDC_connected()) {
+      return size - rest;
```

Comment:
Should be better to use break in this case to exit while then `return size -rest`.
```suggestion
      break;
```

Reply:
done